cmd/backup-manager/app/util/backup_size.go

```diff
@@ -213,24 +213,52 @@ func calculateSnapshotSize(volumeId, snapshotId string) (uint64, uint64, error)
 
 	var nextToken *string
 	for {
-		resp, err := ebsSession.EBS.ListSnapshotBlocks(&ebs.ListSnapshotBlocksInput{
-			SnapshotId: aws.String(snapshotId),
-			MaxResults: aws.Int64(ListSnapMaxReturnResult),
-			NextToken:  nextToken,
-		})
-		numApiReq += 1
-		if err != nil {
-			return 0, numApiReq, err
+		// Each retry interval is around 1 second, and no more than 60 times retry (~1 minute)
+		backoff := wait.Backoff{
+			Duration: time.Second,
+			Steps:    60,
+			Factor:   1.0,
+			Jitter:   0.1,
 		}
-		if resp.BlockSize != nil {
-			snapshotSize += uint64(len(resp.Blocks)) * uint64(*resp.BlockSize)
+
+		isAllListed := false
+
+		listBlocks := func() error {
+			resp, err := ebsSession.EBS.ListSnapshotBlocks(&ebs.ListSnapshotBlocksInput{
+				SnapshotId: aws.String(snapshotId),
+				MaxResults: aws.Int64(ListSnapMaxReturnResult),
+				NextToken:  nextToken,
+			})
+			numApiReq += 1
+			if err != nil {
+				return err
+			}
+			if resp.BlockSize != nil {
+				snapshotSize += uint64(len(resp.Blocks)) * uint64(*resp.BlockSize)
+			}
+
+			// check if there is more to retrieve
+			if resp.NextToken == nil {
+				isAllListed = true
+			}
+			nextToken = resp.NextToken
+
+			return nil
 		}
 
-		// check if there is more to retrieve
-		if resp.NextToken == nil {
+		isRetry := func(err error) bool {
```

Comment:
Maybe we can use the method `IsErrorThrottle` from aws sdk  https://github.com/aws/aws-sdk-go/blob/main/aws/request/retryer.go#L230

Reply:
Fixed